examples/basic_operations.js

```diff
@@ -1,27 +1,42 @@
 "use strict";
 
-var Redis = require("ioredis");
-var redis = new Redis();
+const Redis = require("ioredis");
+const redis = new Redis(process.env.redisPort, process.env.redisEndpoint, {password: process.env.redisPW});
 
 // ioredis supports all Redis commands:
-redis.set("foo", "bar");
+redis.set("foo", "bar"); // returns promise which resolves to string, "Ok"
+
+// the format is: redis[SOME_REDIS_COMMAND_IN_LOWERCASE](ARGUMENTS_ARE_JOINED_INTO_COMMAND_STRING)
+// the js: ` redis.set("mykey", "Hello") ` is equivalent to the cli: ` redis> SET mykey "Hello" `
+
+// ioredis supports the node.js callback style
 redis.get("foo", function(err, result) {
   if (err) {
     console.error(err);
   } else {
-    console.log(result);
+    console.log(result); // Promise resolves to "bar"
   }
 });
+
 redis.del("foo");
 
-// Or using a promise if the last argument isn't a function
+// Or ioredis returns a promise if the last argument isn't a function
 redis.get("foo").then(function(result) {
   console.log(result);
 });
 
 // Arguments to commands are flattened, so the following are the same:
 redis.sadd("set", 1, 3, 5, 7);
 redis.sadd("set", [1, 3, 5, 7]);
+redis.spop("set"); // Promise resolves to "5" or another item in the set
+
+// Most responses are strings, or arrays of strings
+redis.zadd("sortedSet", 1, "one", 2, "dos", 4, "quatro", 3, "three")
+console.log(redis.zrange("sortedSet", 0, 2, "WITHSCORES")); // Promise resolves to ["one", "1", "dos", "2", "three", "3"] as if the command was ` redis> ZRANGE sortedSet 0 2 WITHSCORES `
```

Comment:
We don't need the `console.log` here since it won't print the actual response. A comment for what will be resolved would be enough.